docs/release_process.md

```diff
@@ -0,0 +1,25 @@
+# Release process
+
+## Frequency
+
+Open Service Mesh will target a release every quarter. These releases will be targeted in the first half of February, May, August, and November. These target release months are subject to change if a release is pushed. In other words, if a release misses February and is pushed out in March then the next release will target June instead of May.
+
+## Process for requesting a change
+
+If you would like to see a new feature added to Open Service Mesh, please [open an issue](https://github.com/openservicemesh/osm/issues). This issue will be discussed and, if approved, prioritized in the backlog.
+
+## Creating a new release
+
+When there is a new release being planned and developed, a [milestone](https://github.com/openservicemesh/osm/milestones) will be created. This new milestone will have issues added to it that will be prioritized from the [vFuture milestone](https://github.com/openservicemesh/osm/milestone/32). Once the release milestone is complete, a new release of Open Service Mesh will be created and this milestone will be closed. At that moment, the next release milestone will be created and items from vFuture will be added.
```

Comment:
I think it makes sense to have at least 2-4 future release milestones in existence at once, as opposed to one at a time. It provides more clarity for users, and allows for better planning.

Reply:
What would be the differences between the multiple future release milestones? What's the benefit in splitting up approved work that isn't prioritized yet?

Reply:
it allows us to prioritize and plan past a 3 month horizon, while giving users some clarity on what's to come. 

I think it makes the most sense when we add larger work items (from comment above), to a release, but maybe with a "release-candidate" label, meaning its not a blocker.

Reply:
I think that extra granularity doesn't add enough value to compensate the additional overhead of maintaining that. But perhaps in the future if we feel like that is a gap then we should modify accordingly. I think it is too much on the first pass of this without really having a clear need of that. But I'll defer to other maintainers of the project. // @shashankram @snehachhabria @nojnhuh @jaellio

Reply:
+1 @trstringer.

Reply:
I don't think we need to always have a set number of upcoming version milestones. We can always have at least one and if we think it would be worth it to have more we can create them as needed. I suspect committing to always having 3 or 4 would be a pain, but giving ourselves a chance to do some extra planning further out wouldn't hurt.